manifests/0000_90_ingress-operator_03_prometheusrules.yaml

```diff
@@ -0,0 +1,35 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: ingress-operator
+  namespace: openshift-ingress-operator
+  labels:
+    role: alert-rules
+spec:
+  groups:
+    - name: openshift-ingress.rules
+      rules:
+      - alert: HAProxyHighRoute5xxResp
+        expr: sum by (route) (irate(haproxy_server_http_responses_total{code="5xx", namespace=~"openshift.+"}[5m])) / sum by (route) (irate(haproxy_server_http_responses_total{namespace=~"openshift.+"}[5m])) * 100 > 5
+        for: 5m
+        labels:
+          severity: warning
+        annotations:
+          message: |
+            "HAProxy high HTTP 5xx error rate on route {{ $labels.route }} (> 5%)"
+      - alert: HAProxyPendingRequests
+        expr: sum by (backend) (haproxy_backend_current_queue) > 0
+        for: 5m
+        labels:
+          severity: warning
+        annotations:
+          message: |
+            "HAProxy requests pending on {{ $labels.backend }} backend"
+      - alert: HAProxyServerConnectionErrors
+        expr: sum by (server) (rate(haproxy_server_connection_errors_total[1m])) * 100 > 5
```

Comment:
Does this include client-side errors, or even things like probes (where a client opens a connection and then closes it, causing an EOF)?  Could this indicate that the application pod (and not HAProxy itself) is having a high error rate?  That might be too noisy for multi-tenant clusters such as OpenShift Online.

Reply:
I did not think of this. For now I think I will remove this alert given the possibility of it being easily triggered (Looking at the HAProxy docs mentions that even malformed requests are factored into server error counts).